src/qibo/models/tsp.py

```diff
@@ -0,0 +1,172 @@
+from qibo.symbols import X, Y, Z
+from qibo.models import Circuit
+from qibo.core.hamiltonians import Hamiltonian, SymbolicHamiltonian
+from qibo import gates
+import numpy as np
+from collections import defaultdict
+
+
+def calculate_two_to_one(num_cities):
+    """
+    Args:
+        num_cities: number of cities
+
+    Returns:
+        a matrix to faciliate conversion from 2 dimension to one dimension
+    """
+    return np.arange(num_cities ** 2).reshape(num_cities, num_cities)
+
+
+def tsp_phaser(distance_matrix, dense=True):
+    """
+    Args:
+        distance_matrix: a numpy matrix encoding the distances
+        dense: whether the hamiltonian is dense,
+
+    Returns: a hamiltonian representing the TSP(traveling salesman problem) phaser according to
```

Comment:
You should add a newline after `Returns:` otherwise it will not render properly in the doc.
You shoud do exactly what you did above for `Args:`.

Reply:
Thanks. I have added the newslines.